packages/shared/src/components/CharacterStats/CharactersOverview.tsx

```diff
@@ -0,0 +1,199 @@
+      </text>
+    </g>
+  );
+};
+
+const RatingLabel = (props: Props) => {
+  if (props.value === 0) return null;
+  return (
+    <g className="recharts-layer recharts-polar-angle-axis-tick">
+      <text
+        className="recharts-text recharts-polar-angle-axis-tick-value"
+        fill="#fff"
+        type="category"
+        cx={props.cx}
+        cy={props.cy}
+        orientation="outer"
+        radius={props.radius}
+        x={props.x}
+        y={props.y}
+        textAnchor={props.textAnchor}
+      >
+        <tspan x={props.x} dy="0em">
+          {props.value}
+        </tspan>
+      </text>
+    </g>
+  );
+};
+
+export const CharactersOverview = (props: { characters: UserCharacterInfo[] }) => {
+  const latestRatingPerClass = useMemo(() => {
+    const data = _.map(
+      _.groupBy(
+        props.characters.flatMap((c) =>
+          c.bracketStats.map((s) => ({ ...s, classId: Utils.getSpecClass(c.specId as CombatUnitSpec) })),
+        ),
+        'classId',
+      ),
+      (bracketStats, classId) => {
+        return {
+          classId,
+          rating2v2:
+            _.max(
+              _.map(
+                bracketStats.filter((s) => s.bracket === '2v2'),
+                (s) => s.latestRating,
+              ),
+            ) ?? 0,
+          rating3v3:
+            _.max(
+              _.map(
+                bracketStats.filter((s) => s.bracket === '3v3'),
+                (s) => s.latestRating,
+              ),
+            ) ?? 0,
+          ratingShuffle:
+            _.max(
+              _.map(
+                bracketStats.filter((s) => s.bracket === 'Rated Solo Shuffle'),
+                (s) => s.latestRating,
+              ),
+            ) ?? 0,
+        };
+      },
+    );
+
+    const missingClasses = new Set<CombatUnitClass>();
+    _.values(CombatUnitClass).forEach((k) => {
+      if (typeof k === 'string') return;
+      missingClasses.add(k);
+    });
+    data.forEach((d) => missingClasses.delete(parseInt(d.classId) as CombatUnitClass));
+    missingClasses.delete(CombatUnitClass.None);
+
+    missingClasses.forEach((classId) => {
+      data.push({ classId: classId.toFixed(), rating2v2: 0, rating3v3: 0, ratingShuffle: 0 });
+    });
+
+    // Group characters that did shuffles first, then 3v3, then 2v2.
+    // Within each group, sort by classId.
+    // This is to minimize overlapping areas between the radars while doing best effort
+    // to maintain a consistent order.
+    return data.sort((a, b) => {
+      return (
+        parseInt(b.classId) +
+        (b.ratingShuffle ? 100000 : b.rating3v3 ? 10000 : b.rating2v2 ? 1000 : 0) -
+        (parseInt(a.classId) + (a.ratingShuffle ? 100000 : a.rating3v3 ? 10000 : a.rating2v2 ? 1000 : 0))
+      );
+    });
+  }, [props.characters]);
+
+  return (
+    <div className="flex flex-col flex-1 min-h-full">
+      <ResponsiveContainer>
+        <RadarChart data={latestRatingPerClass}>
+          <PolarGrid opacity={0.5} />
```

Comment:
Maybe have grid shown at achievement breakpoints?

Reply:
I was trying to do that but Recharts didn't make it easy... will try again when I get a chance.